master/internal/stream/system.go

```diff
@@ -0,0 +1,700 @@
+package stream
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/labstack/echo/v4"
+	"github.com/lib/pq"
+	log "github.com/sirupsen/logrus"
+
+	detContext "github.com/determined-ai/determined/master/internal/context"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/stream"
+	"github.com/determined-ai/determined/master/pkg/syncx/errgroupx"
+)
+
+// JSONB is the golang equivalent of the postgres jsonb column type.
+type JSONB interface{}
+
+const (
+	minReconn      = 1 * time.Second
+	maxReconn      = 10 * time.Second
+	trialChan      = "stream_trial_chan"
+	metricChan     = "stream_metric_chan"
+	experimentChan = "stream_experiment_chan"
+	checkpointChan = "stream_checkpoint_chan"
+	projectChan    = "stream_project_chan"
+)
+
+// PublisherSet contains all publishers, and handles all websockets.  It will connect each websocket
+// with the appropriate set of publishers, based on that websocket's subscriptions.
+//
+// There is one PublisherSet for the whole process.  It has one Publisher per streamable type.
+type PublisherSet struct {
+	DBAddress string
+	Projects  *stream.Publisher[*ProjectMsg]
+	// Experiments *stream.Publisher[*ExperimentMsg]
```

Comment:
That's a good point, I'm in favor of removing them given that we're holding onto the original feature branch to revisit post modular execution

Reply:
The reason I wrote those was to show a pattern.

I wanted to make it as easy as possible for people to see what the pattern was so that the next person could see those comments and pick up on the pattern.

If we had more than one type in what we are landing then I think it would make less sense, because the pattern would already be obvious.

I suppose I'm not opposed to removing the comments if we know who is going to add the non-project types?  After the reorg, @corban-beaird are you still going to be working on this?  Or do you know what the plan is?

Reply:
I'll be on experiment tracking, so I'll be closely tied in with the web team. Being that one of the primary uses cases for this is WebUI, I'm happy to keep ownership of Streaming Updates post-reorg.